docs/src/content/docs/guides/connectionoptions.mdx

```diff
@@ -0,0 +1,188 @@
+
+```javascript
+{
+  "systemPrompt": "You are a helpful assistant...",
+  ...
+  "medium": {
+    "twilio": {}
+  },
+  "initiator": "INITIATOR_AGENT"
+}
+```
+
+Ultravox will return a `joinUrl` that can then be used with Twilio for outgoing or incoming calls.
+
+### Outgoing Calls
+
+It only takes two steps to make an outgoing call to regular phone numbers through Twilio:
+<Steps>
+1. **Create an Ultravox Call** → Create a new call (see [above](#creating-a-phone-call-with-twilio)), and get a `joinUrl`.
+
+1. **Initiate Twilio Phone call** → Use the `joinUrl` with a Twilio [`<Stream>`](https://www.twilio.com/docs/voice/twiml/stream).
+
+    ```javascript
+    // Example using the twilio node library
+    const call = await client.calls.create({
+        twiml: `<Response>
+                    <Connect>
+                        <Stream url="${joinUrl}"/>
+                    </Connect>
+                </Response>`,
+        to: phoneNumber, // the number you are calling
+        from: twilioPhoneNumber // your twilio number
+    });
+    ```
+</Steps>
+
+See the [twilio-outgoing-call](https://github.com/fixie-ai/ultradox/tree/main/examples/twilio-outgoing-call) example for more.
+
+This example shows one of the many options Twilio provides for making outgoing calls. Consult the [Twilio docs](https://www.twilio.com/docs) for more details.
+
+### Incoming Calls
+Incoming calls require essentially the same two steps as outgoing calls:
+
+<Steps>
+1. **Create an Ultravox Call** → Create a new call (see [above](#creating-a-phone-call-with-twilio)), and get a `joinUrl`. *Note: for incoming calls you will want to keep `initiator` set to the default ("user").*
+
+1. **Receive Inbound Twilio Phone call** → Use the `joinUrl` with a Twilio [`<Stream>`](https://www.twilio.com/docs/voice/twiml/stream).
+
+    ```xml
+    <!-- Example using a TwiML Bin -->
+    <?xml version="1.0" encoding="UTF-8"?>
+    <Response>
+        <Connect>
+            <Stream url="your_ultravox_join_url" />
+        </Connect>
+    </Response>
+
+    ```
+</Steps>
+
+The above shows how to create a TwiML Bin and use that for handling the inbound call. Consult the [Twilio docs](https://www.twilio.com/docs) for more on all the options Twilio provides for handling phone calls.
+
+
+
+
+
+
+
+
+
+
+
+
+
+## WebSocket Integration
+
+:::caution[Server-to-Server Only]
+WebSocket connections are designed for server-to-server communication. For browser or mobile applications, use our client SDKs with WebRTC for optimal performance. WebSocket connections over TCP can experience audio blocking and ordering constraints that make them unsuitable for direct client use.
+:::
+
+### Use Cases and Limitations
+#### Ideal Use Cases:
+
+* Server-to-server integrations where you already maintain client connections
+* Scenarios requiring simpler connection handling without WebRTC complexity
+* Applications that need to keep AI generations on your server
```

Comment:
```suggestion
```

Generations happen on our inference server either way.

Reply:
removed this per your other comment